pkg/antctl/runtime/runtime.go

```diff
@@ -47,10 +48,24 @@ func ResolveKubeconfig(path string) (*rest.Config, error) {
 			path = clientcmd.RecommendedHomeFile
 		}
 	}
-	if _, err = os.Stat(path); path == clientcmd.RecommendedHomeFile && os.IsNotExist(err) {
-		return rest.InClusterConfig()
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		if path == clientcmd.RecommendedHomeFile {
+			config, inClusterErr := rest.InClusterConfig()
+			if inClusterErr == nil {
+				return config, nil
+			}
+			return nil, fmt.Errorf(
+				"failed to resolve kubeconfig: neither a valid kubeconfig file was found at '%s', nor could InClusterConfig be used: %w",
+				path, inClusterErr,
+			)
+		}
+		return nil, fmt.Errorf("failed to resolve kubeconfig: kubeconfig file does not exist at path '%s'", path)
+	}
```

Comment:
I believe the incluster way should be tried only when the path is the default one:
```suggestion
	if _, err := os.Stat(path); path == clientcmd.RecommendedHomeFile && os.IsNotExist(err) {
			config, inClusterErr := rest.InClusterConfig()
			if inClusterErr == nil {
				return config, nil
			}
			return nil, fmt.Errorf(
				"failed to resolve kubeconfig: no valid kubeconfig file was found at the default path '%s', and InClusterConfig can't be used: %w",
				path, inClusterErr,
			)
	}
```

Reply:
so my understanding is that there is no change in behavior, just better error messages?

Reply:
> so my understanding is that there is no change in behavior, just better error messages?

There is only one case where a non-existent non-default path can lead to an error message and stop further processing, which is not covered in the original code. Other changes indeed aim to improve the error messaging.